### PR TITLE
Fix elastic search index name for ExecutionStatus

### PR DESCRIPTION
**Pull request by samarabbas**
**What changed?**
Update index key definition for Elastic Search in development_es.yaml to reflect the recent renaming from 
`Status` to `ExecutionStatus`.

**Why?**
Recently we did massive renaming of ElasticSearch indexes.  As part of this renaming `Status` index
created by the system to represent status of closed workflow execution was renamed to
`ExecutionStatus`.  Default in the code is updated to represent this change but dynamic config
override used by local development with ES support was not updated.  This resulted in indexes to
go out of sync in all places which relies on development_es.yaml dynamic config file.  This broke
local development setup and anyone using docker-compose_es.yaml.  

**How did you test it?**
Locally run Temporal server using dynamic config overrides in development_es.yaml.  Made sure
visibility is working after execution closes.

**Potential risks**
This should only impact local development setups with Elastic Search.

### CODE REVIEW

File: config/dynamicconfig/development_es.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ frontend.validSearchAttributes:
       StartTime: "Int"
```
Comment (mastermanu) [inline config/dynamicconfig/development_es.yaml:13]: the index_template.json has this as a long. Same with executiontime and closetime

File: config/dynamicconfig/development_es.yaml [hunk 2]
```diff
@@ -31,7 +33,6 @@ frontend.validSearchAttributes:
       CustomNamespace: "Keyword"
```
Comment (mastermanu) [inline config/dynamicconfig/development_es.yaml:33]: not seeing this in index_template.json

File: config/dynamicconfig/development_es.yaml (hunk 1 above)
Comment (samarabbas) [reply config/dynamicconfig/development_es.yaml:13]: Looks like proto definition only defines the following four types:
```
message Field {
    temporal.server.api.enums.v1.FieldType type = 1;
    oneof data {
        string string_data = 2;
        int64 int_data = 3;
        bool bool_data = 4;
        bytes binary_data = 5;
    }
}
```

Comment (mastermanu) [reply config/dynamicconfig/development_es.yaml:13]: hmm, then maybe we should just fix up index_template to use integer then

Comment (samarabbas) [reply config/dynamicconfig/development_es.yaml:13]: Updated.

**Pull request merged**